test/test_gen_backend_stubs.py

```diff
@@ -0,0 +1,99 @@
+import os
+import tempfile
+import subprocess
+
+from torch.testing._internal.common_utils import TestCase, run_tests
+
+path = os.path.dirname(os.path.realpath(__file__))
+gen_backend_stubs_path = os.path.join(path, '../tools/codegen/gen_backend_stubs.py')
+
+yaml_with_errors = {
+    # Missing key (backend)
+    '''\
+cpp_namespace: torch_xla
+supported:
+- abs
+    ''': 'AssertionError: You must provide a value for "backend"',
```

Comment:
Use expect tests! Future test writers will thank you! You can see a pile of examples of expect tests in test/test_dispatch.py

Reply:
Also, you're not saving that much boilerplate by having a dictionary of YAML with errors. Better to give each a separate test so that you can run tests individually if they fail.

Reply:
Agreed that putting each test yaml in its own test will be way clearer.

Haha so after struggling with expect tests for a bit I also realized that with my current setup it'll fail CI, because the *full* output includes a stack trace containing paths specific to my filesystem. That doesn't seem sustainable, because even if all CI machines run the tests from the same directory, that'll be different from our individual filesystems (/scratch/hirsheybar/.... in my case).

Rather than try to parse out and remove the stack trace, I think the whole thing will be less fragile if I just refactor `gen_backend_stubs.py` to give it a clear entry point function, and call that function directly from the tests.

Downside: it's kind of a less faithful test since it's not "directly" running `gen_backend_stubs.py` and makes minor assumptions about it's internals, but I think that'll make these tests easier to work with. Does that all sound reasonable to you?

Reply:
This is fine, and will also make it easier to attach pdb to debug tests (rather than chasing a command line argument). However, parsing is not completely out of the question either; we do it for cpp stack traces here, for example:

```
    def assertMultiLineEqualMaybeCppStack(self, expect, actual, *args, **kwargs):
        self.assertGreaterEqual(len(actual), len(expect), *args, **kwargs)
        if hasattr(self, "assertMultiLineEqual"):
            self.assertMultiLineEqual(expect, actual[:len(expect)], *args, **kwargs)
        else:
            self.assertEqual(expect, actual[:len(expect)], *args, **kwargs)
        if len(actual) > len(expect):
            cpp_stacktrace_header = "\nException raised from"
            end_header = len(expect) + len(cpp_stacktrace_header)
            self.assertEqual(actual[len(expect): end_header], cpp_stacktrace_header)
```

In general, the price you have to pay for getting to use expect tests is you have to ensure, one way or another, that the output you're expecting on is deterministic. This has usually be worth it, in my experience.